### PR TITLE
[Doc] Add SAP Cloud SDK version requirements

### PR DESCRIPTION
**Pull request by newtork**
## Context

Related backlog item:
* https://github.tools.sap/AI/ai-sdk-java-backlog/issues/106

Related automatic testing:
* #161

Tested via
```
mvn clean package spring-boot:run -pl :spring-app -Dcloud-sdk.version=5.6.0 -Denforcer.skip=true
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,7 +45,9 @@ To use the SAP AI SDK for Java, the following general prerequisites must be met:
     - [How to enable the AI Core service](https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/initial-setup)
 - **SAP AI Core Credentials** to access the AI Core service.
     - [Connecting to SAP AI Core](#connecting-to-sap-ai-core)
-- **(Optional) Spring Boot** version 3 or higher if you are using Spring Boot.
+- **(Optional) Spring Boot** version 3 or higher.
+- **(Optional) SAP Cloud SDK** version `5.6.0` or higher.
```
Comment (MatKuhr) [inline README.md:49]: It's not really optional. What is optional is the user defining a version for it. But contrary to spring boot or cap, it has to be there. 

Personally, I like how cap does this here: https://cap.cloud.sap/docs/java/versions#minimum-versions

**Review by MatKuhr — COMMENTED** [review]
How does 5.6.0 not break with the apache changes? The we are setting the http client version explicitly in the ai sdk, so how does this not clash?

File: README.md [hunk 2]
```diff
@@ -45,7 +45,9 @@ To use the SAP AI SDK for Java, the following general prerequisites must be met:
...
+- **(Optional) SAP CAP SDK** version `3.0.0` or higher.
```
Comment (MatKuhr) [inline README.md:50]:
```suggestion
- **(Optional) SAP CAP Framework** version `3.0.0` or higher.
```

**Comment by newtork** [comment]
> We are setting the http client version explicitly in the AI SDK.

If I see correctly, there is no explicit version. We just declare the dependency explicitly for code quality purposes, since we interact with the API directly. I.e. not make "enforcer" plugin angry.

> so how does this not clash?

In AI SDK we are not using Apache API that was subject to changes over the versions.

File: README.md (hunk 1 above)
Comment (newtork) [reply README.md:49]: "Optional" in the sense of custom version declaration, but sure I can phrase it differently.

File: README.md [hunk 3]
```diff
@@ -37,15 +37,21 @@ The SDK simplifies the setup and interaction with SAP AI Core, allowing you to f
...
-- **(Optional) Spring Boot** version 3 or higher if you are using Spring Boot.
+
+The following table lists the required versions, based on the latest release:
+
+| Dependency | Minimum Version | Recommended Version |
+| --- | --- | --- |
+| JDK | 17 (LTS) | 21 (LTS) |
+| Maven | 3.9 | latest |
+| Spring Boot | 3.0 | latest |
+| SAP Cloud SDK | 5.6.0 | latest |
```
Comment (MatKuhr) [inline README.md:54]: This makes it seem Maven and Spring Boot would be required, but they aren't. 

```suggestion
| SAP Cloud SDK | 5.6.0 | latest |
| (optional) Spring Boot | 3.0 | latest |
```

**Review by MatKuhr — APPROVED** [review]

**Pull request merged**